DKP Insights first draft (#4225)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/insights/index.md

---
layout: layout.pug
navigationTitle: DKP Insights

> title: DKP Insights

Reviewer: I am inclined with DKP Insights but need to run it by Deepak and other key people.

Author: @kanna823 -we touched on that yesterday here: https://mesosphere.slack.com/archives/C02P8A08ZM3/p1648736119497329

excerpt: Use DKP Insights to monitor and tune your environment
menuWeight: 55
beta: false
enterprise: false
techPreview: true
---

<p class="message--note"><strong>NOTE:</strong> We are offering DKP Insights with this release as a Technical preview only. </p>

> We designed DKP Insights to assist Kubernetes Administrators with daily tasks, such as ensuring your clusters are secure, debugging, monitoring, and anticipating issues. DKP Insights collects logs, events, and metrics from the DKP Management cluster and managed or attached Kubernetes clusters. It then generates rules-based insights on potential problems of varying critical levels so that you can quickly identify and resolve them yourself.

Reviewer: This one needs rewording.

Reviewer: How about this: "such as ensuring clusters are secure, resources are optimally used, workloads and clusters follow best practices, and resolve any problems that arise"

Reviewer 2: I agree it needs rewording.    But we need to work on the proposed reformulation, because all the phrases need to agree with the verb here, which is 'ensuring'; the phrase 'resolve any problems that arise' doesn't agree.

But I also worry about over-promising here.  Insights can't really 'ensure' anything - it can only surface some issues and the user still has to resolve them themselves.  So I don't think we should say 'ensuring clusters are secure' because this implies to me that insights can 'ensure all possible security issues are surfaced'.

How about this:

```suggestion
We designed DKP Insights to assist Kubernetes Administrators with daily tasks, such as checking for security issues, monitoring resource usage, verifying that workloads and clusters follow best practices, and resolving common issues. DKP Insights collects logs, events, and metrics from the DKP Management cluster and managed or attached Kubernetes clusters. It then generates rules-based insights on potential problems of varying critical levels so that you can quickly identify and resolve them yourself.
```

Author: All solid points, thanks @cneth 
I also wonder about the using a persona here: Kube Admins. While in practice this is something good that we want to get into with the docs eventually, we haven't yet used personas in the technical doc and don't want to confuse anyone. 
Is this generally understood and won't throw any users off, or would the ambiguous "you" be better here?
So:
"We designed DKP Insights to assist **you** with..."
It seems minor, but we want to avoid any mental breaks or pauses when a user reads this; ie, don't want them stopping mid-sentence asking "are they talking about me?".

Author: @cneth - I forgot to commit this before pulling back into VSC. I have this added and it'll show up in a minute when I push this back up from VSC.

DKP Insights works seamlessly within networked or air-gapped environments. It collects metrics, logs, and events from multiple data sources and uses rule-based heuristics to detect issues from the data and generate insight items for the users.

You can filter the insight items for a selected cluster or project by:

> 

Author: @nataliepina - edited this list according to your suggestions.

- Project name
- Cluster name
- Description

> You can also toggle into each severity level.

Author: ```suggestion
From the DKP Dashboard Insights table, you can also toggle into each severity level:

- Critical
- Warning
- Notice/informational
```

---
layout: layout.pug
navigationTitle: DKP Insights
title: DKP Insights
excerpt: Use DKP Insights to monitor and tune your environment
menuWeight: 55
beta: false

> enterprise: false

Reviewer 3: More so curious how this works since I'm new to docs. Can you remove the beta: false and enterprise: false? I'm not sure if that has any other implications but I thought those were just for the badges? If you can, it would be nice to keep it clean if you don't need these explicitly set to false.

techPreview: true
---

> <p class="message--note"><strong>NOTE:</strong> We are offering DKP Insights with this release as a Technical preview only. </p>

Reviewer 3: The casing of Technical preview feels off to me? Should this be "Technical Preview"? Or lowercase?

We designed DKP Insights to assist Kubernetes Administrators with daily tasks, such as checking for security issues, monitoring resource usage, verifying that workloads and clusters follow best practices, and resolving common issues. DKP Insights collects logs, events, and metrics from the DKP Management cluster and managed or attached Kubernetes clusters. It then generates rules-based insights on potential problems of varying critical levels so that you can quickly identify and resolve them yourself.

DKP Insights works seamlessly within networked or air-gapped environments. It collects metrics, logs, and events from multiple data sources and uses rule-based heuristics to detect issues from the data and generate insight items for the users.

You can filter the insight items for a selected cluster or project by:

- Project name
- Cluster name
- Description

> From the DKP Workspace Dashboard, you can toggle by severity level:

Reviewer 3: You cannot toggle from the Workspace Dashboard, we only have the card displaying the top 3 insights there. You can toggle from the same insights summary page referenced in the filtering above.

Reviewer 3: Should this be "DKP Insights Dashboard"?

- Critical
- Warning
- Notice/informational

> Alternately, the DKP Insights dashboard gives you several different ways to filter and sort insight itmes. Access the DKP Insights dashboard by selecting Insights from the sidebar menu.

Reviewer 3: ```suggestion
Alternately, the DKP Insights dashboard gives you several different ways to filter and sort insight items. Access the DKP Insights dashboard by selecting Insights from the sidebar menu.
```

---
layout: layout.pug
navigationTitle: DKP Insights
title: DKP Insights
excerpt: Use DKP Insights to monitor and tune your environment
menuWeight: 55
beta: false
enterprise: false
techPreview: true
---

<p class="message--note"><strong>NOTE:</strong> We are offering DKP Insights with this release as a Technical preview only.</p>

DKP Insights assists Kubernetes Administrators with routine tasks:

- Resolving common issues
- Monitoring resource usage
- Checking security issues
- Verifying workloads and clusters follow best practices

Access the DKP Insights Dashboard by selecting **Insights** from the sidebar menu.

DKP Insights consists of two components:

- DKP Insights Management: Runs on the Kommander [Management](../clusters/management-cluster) cluster.

> - DKP Insights Engine: Runs on each [Attached](../clusters/) Kubernetes cluster.

Author: ```suggestion
- DKP Insights Engine: Runs on each [Attached or Managed](../clusters/) Kubernetes cluster.
```
